### PR TITLE
Updated to NServiceBus.RabbitMQ 6.0.0

### PR DESCRIPTION
**Pull request by ramonsmits**
Closes https://github.com/Particular/ServiceControl/pull/2077

I've chosen to not break the connection string parsing as that would result in a breaking change.

### CODE REVIEW

**Comment by bording** [comment]
> I've chosen to not break the connection string parsing as that would result in a breaking change.

Not sure what you mean by this. The way we implemented the connection string parsing in the transport let's both formats be used. We should definitely copy the update code over so that both formats can be used in ServiceControl as well.

**Comment by bording** [comment]
Of course, looking more at the code in here, there are a lot of assumptions around the connection string format baked in here, so there's going to be more work than I initially thought to enable both of them. 😢 

I've at least gone ahead and updated the classes copied from the transport with the latest versions to ensure we are keeping things like default values in sync.

For now, it doesn't seem like we could actually do the additional work to enable the amqp connection string format because of ServiceControl's increasingly outdated assumption that all configuration is passed in via a connection string.

File: src/ServiceControl.Transports.RabbitMQ/ConnectionConfiguration.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ class ConnectionConfiguration
         const string defaultVirtualHost = "/";
         const string defaultUserName = "guest";
         const string defaultPassword = "guest";
-        const ushort defaultRequestedHeartbeat = 5;
+        const ushort defaultRequestedHeartbeat = 60;
```
Comment (tmasternak) [inline src/ServiceControl.Transports.RabbitMQ/ConnectionConfiguration.cs:21]: Why did we change this?

Comment (bording) [reply src/ServiceControl.Transports.RabbitMQ/ConnectionConfiguration.cs:21]: This was changed in https://github.com/Particular/NServiceBus.RabbitMQ/pull/608 to align our default value with the client's actual default. Five seconds was way too short, and we should never have been setting it that low.

**Pull request merged**